cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -2659,17 +2659,41 @@
 ;; f32 or f64 despite the source perhaps being an integer vector since the
 ;; result of the instruction is the same.
 (rule 2 (lower (store flags
-                    (has_type (ty_32 _) (extractlane value (u8_from_uimm8 0)))
+                    (has_type $F32 (extractlane value (u8_from_uimm8 0)))
                     address
                     offset))
       (side_effect
        (x64_movss_store (to_amode flags address offset) value)))
-(rule 3 (lower (store flags
-                    (has_type (ty_64 _) (extractlane value (u8_from_uimm8 0)))
+(rule 2 (lower (store flags
+                    (has_type $F64 (extractlane value (u8_from_uimm8 0)))
                     address
                     offset))
       (side_effect
        (x64_movsd_store (to_amode flags address offset) value)))
+(rule 2 (lower (store flags
+                    (has_type $I8 (extractlane value (u8_from_uimm8 n)))
```

Comment:
I checked and `u8_from_uimm8` does not guarantee that `n` is in the right range (e.g., 0 to 15 here). Did you see any validation that would prevent wrong `n` here at the CLIF level or do we rely exclusively on the Wasm-level construction? If there is nothing and it ends up being too tricky to do here in ISLE, maybe `emit.rs` could gain some `assert!`s or something like that.

Reply:
I think [this is the block](https://github.com/bytecodealliance/wasmtime/blob/9ed441e6575685e3c795ab4a504e7f90be01ce3e/cranelift/codegen/src/verifier/mod.rs#L1663-L1687) in the verifier which validates that lanes are always inbounds, so I think we should be good on that front. I can add some extra asserts though to the backend too.